core/api/service/state/CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ target_link_libraries(state_api_service
     api_service
     trie_storage
     blob
+    binaryen_metadata_api
```

Comment:
we use only interface of metadata api for state api. So I think adding this target is redundant

Reply:
We have MetadataImpl external linkage which can not be resolved by linker.